### PR TITLE
Upgrade sound engine to support 44100hz sample rate

### PR DESCRIPTION
**Pull request by karst**
Cherry Picked from OpenMusic's upcoming update.

### CODE REVIEW

**Comment by ZehMatt** [comment]
I believe I modified this before to stop the crackling issue which was notably observed on the Merry go Round, this might introduce it again.

**Comment by karst** [comment]
I definitely heard an issue with a flac file using this. I forgot to tag @IntelOrca who wrote this code, I just cherry-picked it. 
44100 will definitely be necessary in the future though, will do some testing. Might it be an issue with the audio engine we're using?

**Comment by IntelOrca** [comment]
@karst Which flac file? Was the flac file also 44100? When switching the sample rate, the only extra delay you would get is SDL having to resample the music. The 8x buffer is probably too high. That changes was to try and fix new music which had clicking because the sample rate was far too high. 4096 or 8192 might suffice.

**Comment by ZehMatt** [comment]
If memory serves right the upscaling was pretty bad and the vanilla content uses only 22k.

**Comment by karst** [comment]
> @karst Which flac file? Was the flac file also 44100? When switching the sample rate, the only extra delay you would get is SDL having to resample the music. The 8x buffer is probably too high. That change was to try and fix new music which had clicking because the sample rate was far too high. 4096 or 8192 might suffice.

I have rerendered the song in question into a lower sample rate which seems to help.

> If memory serves right the upscaling was pretty bad and the vanilla content uses only 22k.

I have a high quality assetpack in the works with reworks from the RCT2 songs, which will be in 44100hz, as well as OpenMusic being available in said sample rate.

**Comment by ZehMatt** [comment]
> > @karst Which flac file? Was the flac file also 44100? When switching the sample rate, the only extra delay you would get is SDL having to resample the music. The 8x buffer is probably too high. That change was to try and fix new music which had clicking because the sample rate was far too high. 4096 or 8192 might suffice.
> 
> I have rerendered the song in question into a lower sample rate which seems to help.
> 
> > If memory serves right the upscaling was pretty bad and the vanilla content uses only 22k.
> 
> I have a high quality assetpack in the works with remasters from the RCT2 songs (sent to me by Allister Brimble himself), which will be in 44100hz, as well as OpenMusic being available in said sample rate.

We shouldn't lower it until that is complete and enabled by default, the problem might not be there anymore I'll do some testing later.

File: src/openrct2-ui/audio/AudioMixer.cpp [hunk 1]
```diff
@@ -27,10 +27,10 @@ void AudioMixer::Init(const char* device)
     Close();
 
     SDL_AudioSpec want = {};
-    want.freq = 22050;
+    want.freq = 44100;
     want.format = AUDIO_S16SYS;
     want.channels = 2;
-    want.samples = 2048;
+    want.samples = 2048 * 8;
```
Comment (ZehMatt) [inline src/openrct2-ui/audio/AudioMixer.cpp:33]: I would be okay with 4096, I also experimented a bit with this and there was no difference other than memory consumption but that seems a bit too high.

**Comment by ZehMatt** [comment]
Tested it a bit and there doesn't seem to be any noticeable differences, left one remark however.

**Review by Broxzier — APPROVED** [review]
Didn't notice any strange side effects while playing

**Pull request merged**